tests/schemas/test_user_schema.py

```diff
@@ -10,17 +18,28 @@ def test_unique_email(self, empty_test_db, create_join_staff):
             'email': user.email
         }
 
-        validation_errors = UserSchema().validate(payload)
+        validation_errors = self.schema().validate(payload)
 
         assert 'email' in validation_errors
         assert validation_errors['email'] == [f"A user with email '{user.email}' already exists"]
 
 
-class TestUserValidations(UserSchemaValidations):
-    pass
+class TestUserSchemaValidations(UserSchemaValidations):
+    def setup(self):
+        self.schema = UserSchema
 
 
-class TestUserRegisterSchemaValidations(UserSchemaValidations):
+class TestInheritedUserRegisterSchemaValidations(UserSchemaValidations):
+    def setup(self):
+        self.schema = UserRegisterSchema
```

Comment:
What are these classes doing? Am I missing something, or do these still need to be fleshed out?

If these don't do anything yet, it might be worth throwing in a couple of comments explaining what should go there for the time being, and then opening up a corresponding issue.

Other than that, the changes look good.

Reply:
Thanks for asking @aedwardg ! These are absolutely doing something. They're confirming the contract between the `UserRegisterSchema` and the `UserSchema`. 

In the `schemas/user.py` file there is a `UserSchema` and a `UserRegisterSchema`. The `UserSchema` has a validation for the `email` field that the `UserRegisterSchema` inherits and must validate too.

As a general testing principle, inherited behavior must be tested to ensure the inherited class does nothing to override behavior that should not be overridden. To do this the `UserSchemaValidations` class was created to assert the email validations (and any other validations that get added in the future). This class does not start with the word `Test` so it is not ran by pytest.

The two classes above that appear to do nothing actually executes the tests defined in `UserSchemaValidations` because they include all the methods defined in that class through inheritance. 

Originally I made a mistake when I set this up since the test method in the `UserSchemaValidations` was always calling the `UserSchema`. So now it is set up correctly and injects the schema's we need to use for the email validation test as seen on line 21.

Reply:
Let me know if this explanation helps or not.

Reply:
@NickSchimek: 

So, just to make sure I understand this, `UserSchemaValidations` is not run by pytest because it doesn't start with the word "Test," but since these two classes _do_ start with "Test" and they inherit from `UserSchemaValidations`, pytest runs them and also invokes the inherited `test_unique_email()` method for each using the `self.schema` you provide in each class's `setup()` method?

If that's it, I think I understand, but it definitely feels like I'm in the movie Inception 😄

Reply:
Yep, you got it! That is one of OOPs strongest features. As long as the walls don't start falling down - I think you'll be good.